src/DataSets/Forms/Save.component.js

```diff
@@ -34,11 +35,15 @@ const Save = React.createClass({
     },
 
     _redirectAfterSave() {
+        const {dataset, associations} = this.props.store;
+        log('edit', 'success', dataset);
```

Comment:
You have access to the action being performed (add/edit/clone) in `this.props.store.action`.

(This solves your first item in "known issues")